### PR TITLE
[AFR] Implemented Brazen Dwarf and Feywild Trickster

### PR DESCRIPTION
**Pull request by weirddan455**
Issue #7808 

I know @theelk801 said he was working on the dice mechanic so I just wanted to make sure this was fine for these triggered abilities.  I tested with [[Steel Squirrel]] and it seemed to work correctly.  I wasn't able to test with new cards.  They're giving me a null pointer exception (see issue #7962)

### CODE REVIEW

File: Mage/src/main/java/mage/abilities/common/DiceRolledTriggeredAbility.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    }
+
+    @Override
+    public boolean checkEventType(GameEvent event, Game game) {
+        return event.getType() == GameEvent.EventType.DICE_ROLLED;
+    }
+
+    @Override
+    public boolean checkTrigger(GameEvent event, Game game) {
+        return event.getPlayerId().equals(getControllerId());
```
Comment (JayDi85) [inline Mage/src/main/java/mage/abilities/common/DiceRolledTriggeredAbility.java:35]: All other cards with `DICE_ROLLED` trigger uses additional `event.getFlag()` here (it's a custom param that can be used in events, see example in param's comment). I can't find the real usage of that flag in `ROLL_DICE` replace events, but it must be used same way here or deleted from other DICE_ROLLED's cards as unused.

shot_210703_000201


C:\Users\JayDi\Pictures\CloudShot\shot_210703_000201.png

Comment (theelk801) [reply Mage/src/main/java/mage/abilities/common/DiceRolledTriggeredAbility.java:35]: we're gonna have to do some refactoring anyway like I did with coin flips, so we'll want a `RollDiceEvent` for replacement effects and a `DiceRolledEvent` class for triggers

Comment (theelk801) [reply Mage/src/main/java/mage/abilities/common/DiceRolledTriggeredAbility.java:35]: @weirddan455 if you want to take this assignment then feel free

Comment (weirddan455) [reply Mage/src/main/java/mage/abilities/common/DiceRolledTriggeredAbility.java:35]: Well I guess I'm not sure what data we'll need to store in those classes.  I just saw [[Barbarian Class]] has been spoiled which looks like we'll at least need to support multiple dice roll in a replacement effect and I'm not sure what else is coming.

Comment (theelk801) [reply Mage/src/main/java/mage/abilities/common/DiceRolledTriggeredAbility.java:35]: yeah, that’s the fun of preview season! luckily the set will be fully previewed by wednesday except for the commander decks

Comment (JayDi85) [reply Mage/src/main/java/mage/abilities/common/DiceRolledTriggeredAbility.java:35]: @weirddan455 multi rolls already supported in replace event. See [[The Big Idea]] as example.

Comment (weirddan455) [reply Mage/src/main/java/mage/abilities/common/DiceRolledTriggeredAbility.java:35]: I see.  Why do we need to refactor into new classes then @theelk801 ?

Comment (theelk801) [reply Mage/src/main/java/mage/abilities/common/DiceRolledTriggeredAbility.java:35]: because I didn't look too closely into it and assumed we had to

Comment (weirddan455) [reply Mage/src/main/java/mage/abilities/common/DiceRolledTriggeredAbility.java:35]: Ok, I'm not saying it's a bad idea.  Was just curious.  I haven't looked into it too much either.  Maybe it's best to wait until full set gets spoiled so we know what all needs to be done.

Comment (theelk801) [reply Mage/src/main/java/mage/abilities/common/DiceRolledTriggeredAbility.java:35]: yeah that's basically what I was planning to do

**Comment by theelk801** [comment]
I've merged this branch into the other PR so I'm gonna close this one.

**Pull request closed without merging**